src/test/java/uk/gov/justice/probation/courtcasematcher/messaging/SqsMessageReceiverIntTest.java

```diff
@@ -0,0 +1,141 @@
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.justice.probation.courtcasematcher.model.courtcaseservice.CourtCase;
+import uk.gov.justice.probation.courtcasematcher.model.externaldocumentrequest.Case;
+import uk.gov.justice.probation.courtcasematcher.model.externaldocumentrequest.Info;
+import uk.gov.justice.probation.courtcasematcher.model.offendersearch.SearchResponse;
+import uk.gov.justice.probation.courtcasematcher.service.TelemetryService;
+import uk.gov.justice.probation.courtcasematcher.wiremock.WiremockExtension;
+import uk.gov.justice.probation.courtcasematcher.wiremock.WiremockMockServer;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.matchingJsonPath;
+import static com.github.tomakehurst.wiremock.client.WireMock.putRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static org.awaitility.Awaitility.await;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ActiveProfiles({"test"})
+public class SqsMessageReceiverIntTest {
+
+    private static String singleCaseXml;
+
+    @Autowired
+    private TelemetryService telemetryService;
+
+    private static final WiremockMockServer MOCK_SERVER = new WiremockMockServer(8090);
+
+    @RegisterExtension
+    static WiremockExtension wiremockExtension = new WiremockExtension(MOCK_SERVER);
+
+    @BeforeAll
+    static void beforeAll() throws IOException {
+        final String basePath = "src/test/resources/messages";
+        singleCaseXml = Files.readString(Paths.get(basePath +"/external-document-request-single-case.xml"));
+        MOCK_SERVER.start();
+    }
+
+    @AfterAll
+    static void afterAll() {
+        MOCK_SERVER.stop();
+    }
+
+    private static final String QUEUE_NAME = "crime-portal-gateway-queue";
+
+    @Autowired
+    private QueueMessagingTemplate queueMessagingTemplate;
+
+    @Test
+    public void whenReceivePayload_thenSendCase() {
+
+        queueMessagingTemplate.convertAndSend(QUEUE_NAME, singleCaseXml);
+
+        await()
+            .atMost(5, TimeUnit.SECONDS)
+            .until(() -> countPutRequestsTo("/court/B10JQ/case/1600032981") == 1);
+
+        MOCK_SERVER.verify(
+            putRequestedFor(urlEqualTo("/court/B10JQ/case/1600032981"))
+                .withRequestBody(matchingJsonPath("pnc", equalTo("2004/0012345U")))
+        );
+
+        verify(telemetryService).trackSQSMessageEvent(any(String.class));
+        verify(telemetryService).trackCourtCaseEvent(any(Case.class), any(String.class));
+        verify(telemetryService).trackOffenderMatchEvent(any(CourtCase.class), any(SearchResponse.class));
+        verify(telemetryService).trackCourtListEvent(any(Info.class), any(String.class));
```

Comment:
I am still trying to feel my way into these tests.  Looking at probation-to-prison-update, they didn't mock much because the end result of most things being tested was another call to another REST endpoint. So you can just cover that off with wiremock. I just felt like we could still verify some service calls.